src/Analyzer.JsonRuleEngine/Expressions/AllOfExpression.cs

```diff
@@ -1,59 +1,30 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-using System;
-using System.Collections.Generic;
-using Microsoft.Azure.Templates.Analyzer.Types;
-
 namespace Microsoft.Azure.Templates.Analyzer.RuleEngines.JsonEngine.Expressions
 {
     /// <summary>
     /// Represents an allOf expression in a JSON rule.
     /// </summary>
-    internal class AllOfExpression : Expression
+    internal class AllOfExpression : LogicalExpression
```

Comment:
There's also some duplication of code in the test files for AllOfExpression and AnyOfExpression. These likely can be simplified greatly since AllOfExpression has been simplified so much, so maybe the shared code just make it a new test file for LogicalExpression.

Reply:
Good idea, will do.